pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -448,7 +448,7 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 					},
 					{
 						Alert: "KubeVirtComponentExceedsRequestedMemory",
-						Expr:  intstr.FromString(fmt.Sprintf(`((kube_pod_container_resource_requests{namespace="%s",container=~"virt-controller|virt-api|virt-handler|virt-operator",resource="memory"}) - on(pod) group_left(node) container_memory_usage_bytes{namespace="%s"}) < 0`, ns, ns)),
+						Expr:  intstr.FromString(fmt.Sprintf(`((kube_pod_container_resource_requests{namespace="%s",container=~"virt-controller|virt-api|virt-handler|virt-operator",resource="memory"}) - on(pod) group_left(node) container_memory_working_set_bytes{id=~".*slice",namespace="%s"}) < 0`, ns, ns)),
```

Comment:
@saschagrunert I am sorry for the misleading with the `id~=".*slice"` filtering, but it won't work in the cgroups v2 case. When kubelet is set to work with `driver=cgroupsfs` the naming is different than what its does with systemd units. The best candidate for filtering is using `container=""` because then we filter the accumulated metric for the pod slice. 

I got these snapshots to illustrate what I am talking about:
![1 23-systemd](https://user-images.githubusercontent.com/59094274/160289842-8b679940-5a1a-47c4-be64-210fcec8b2dc.png)

This screenshot was taken from k8s 1.23, please note the id labels, please also note the accumulated metric
doesn't have the `container` label.


![1 23-cgroupsfs](https://user-images.githubusercontent.com/59094274/160289881-28082ada-f1f5-4329-ab8c-4d4117daa92c.png)

This screenshot is from k8s 1.23 with cgroups v2 enabled. Besides the faulty metrics related to the `container="POD"` please note that the id has changed and it doesn't have the `.slice` and `.scope` suffix anymore.

Therefore my thought was to use `container=""` as a filter (I've validated it and it works)

Reply:
@sradco FYI

Reply:
Ah alright changed it as suggested and verified that it also works with OpenShift 4.10.x.